src/seals/base_envs.py

```diff
@@ -23,104 +26,74 @@ class ResettablePOMDP(gym.Env, abc.ABC, Generic[State, Observation, Action]):
     meet these criteria.
     """
 
-    _state_space: gym.Space
-    _observation_space: gym.Space
-    _action_space: gym.Space
-    _cur_state: Optional[State]
+    state_space: spaces.Space[StateType]
+
+    _cur_state: Optional[StateType]
     _n_actions_taken: Optional[int]
 
-    def __init__(
-        self,
-        *,
-        state_space: gym.Space,
-        observation_space: gym.Space,
-        action_space: gym.Space,
-    ):
-        """Build resettable (PO)MDP.
-
-        Args:
-            state_space: gym.Space containing possible states.
-            observation_space: gym.Space containing possible observations.
-            action_space: gym.Space containing possible actions.
-        """
-        self._state_space = state_space
-        self._observation_space = observation_space
-        self._action_space = action_space
+    def __init__(self):
+        """Build resettable (PO)MDP."""
 
         self._cur_state = None
         self._n_actions_taken = None
-        self.seed()
 
     @abc.abstractmethod
-    def initial_state(self) -> State:
+    def initial_state(self) -> StateType:
         """Samples from the initial state distribution."""
 
     @abc.abstractmethod
-    def transition(self, state: State, action: Action) -> State:
+    def transition(self, state: StateType, action: ActType) -> StateType:
         """Samples from transition distribution."""
 
     @abc.abstractmethod
-    def reward(self, state: State, action: Action, new_state: State) -> float:
+    def reward(self, state: StateType, action: ActType, new_state: StateType) -> float:
         """Computes reward for a given transition."""
 
     @abc.abstractmethod
-    def terminal(self, state: State, step: int) -> bool:
+    def terminal(self, state: StateType, step: int) -> bool:
         """Is the state terminal?"""
 
     @abc.abstractmethod
-    def obs_from_state(self, state: State) -> Observation:
+    def obs_from_state(self, state: StateType) -> ObsType:
         """Sample observation for given state."""
 
-    @property
```

Comment:
Why are we deleting these?

I thought `observation_space` and `action_space` were mandatory in Gymnasium? And we probably want to expose `state_space` publicly somewhere

Reply:
We just use the properties provided by the superclass now. This is the canonical way to do it [according to the tutorial](https://gymnasium.farama.org/tutorials/gymnasium_basics/environment_creation/#declaration-and-initialization).